src/map/mob.cpp

```diff
@@ -1444,8 +1444,15 @@ static int mob_ai_sub_hard_slavemob(struct mob_data *md,t_tick tick)
 			}
 		}
 
-		if(md->target_id) //Slave is busy with a target.
-			return 0;
+		if(md->target_id) {//Slave is busy with a target.
+			if (bl->type == BL_PC && md->master_dist > 5) {//Player's slave should come back when master's too far, even if it is doing with a target.
```

Comment:
```suggestion
			// Player's slave should come back when master's too far, even if it is doing with a target.
			if (bl->type == BL_PC && md->master_dist > 5) {
```
Besides that a question: Should this be `( MOB_SLAVEDISTANCE * 2 + 1 )` instead of hardcoding 5? Or even a battle config?

Reply:
Guess battle config sounds good?